src/proxy.rs

```diff
@@ -60,79 +60,63 @@ impl Proxy {
         let handle = thread::spawn(move || {
             let mut buf = [0; 32];
```

Comment:
This will become problematic in future. This is essentially your proxy buffer size which means the app can only ever shift 32 packets of 8 bits at a time. This should ideally be a config value so users can configure their own MTU (message transfer unit) that works best with their network stack/system/server/traffic. For now it simplifies the whole query checking mechanism but it should be changed before being made stable because this may cause problems on a high-traffic server.